lib/srv/db/common/kerberos/kinit/ldap.go

```diff
@@ -0,0 +1,201 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package kinit
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"log/slog"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/go-ldap/ldap/v3"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth/windows"
+)
+
+type ldapConfig struct {
+	Address string
+
+	TLSServerName string
+	TLSCACert     *x509.Certificate
+
+	Domain            string
+	ServiceAccount    string
+	ServiceAccountSID string
+}
+
+type ldapConnectorConfig struct {
+	logger     *slog.Logger
+	authClient windows.AuthInterface
+
+	ldapConfig  ldapConfig
+	clusterName string
+}
+
+type ldapConnector struct {
+	ldapConnectorConfig
+
+	dialLDAPServerFunc func(ctx context.Context) (ldap.Client, error) // only used in tests.
+}
+
+func newLDAPConnector(cfg ldapConnectorConfig) *ldapConnector {
+	if cfg.logger == nil {
+		cfg.logger = slog.Default()
```

Comment:
nit: maybe we can add the domain and service account to logger?

Reply:
Great idea, done.